### PR TITLE
WebVTT API tests

### PR DESCRIPTION
**Pull request by BenjaminSchaaf**
I've reorganised and expanded the api tests for the VTTCue interface.
Folders more closely match the way the spec is organised.
Added help links to all tests to link back to the spec.

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Chrome (unstable channel) #
Testing web-platform-tests at revision fa67745f15d6223bf2b05fa0c40449196f724aef
Using browser at version 58.0.3013.3 dev
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>12 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/4921/webvtt/api/VTTCue/getCueAsHTML.html">/webvtt/api/VTTCue/getCueAsHTML.html</a></summary>


|                  Subtest                  | Results |
|-------------------------------------------|---------|
|                                           | OK      |
| `VTTCue.getCueAsHTML(), <c>`              | PASS    |
| `VTTCue.getCueAsHTML(), creating the cue` | PASS    |
| `VTTCue.getCueAsHTML(), <b>`              | PASS    |
| `VTTCue.getCueAsHTML(), <u>`              | PASS    |
| `VTTCue.getCueAsHTML(), <rt>`             | PASS    |
| `VTTCue.getCueAsHTML(), <v>`              | PASS    |
| `VTTCue.getCueAsHTML(), <i>`              | PASS    |
| `VTTCue.getCueAsHTML(), x\0`              | FAIL    |
| `VTTCue.getCueAsHTML(), <ruby>`           | PASS    |
| `VTTCue.getCueAsHTML(), <c.a.b>`          | PASS    |
| `VTTCue.getCueAsHTML(), <v a b>`          | PASS    |
| `VTTCue.getCueAsHTML(), <00:00:00.500>`   | PASS    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4921/webvtt/api/VTTCue/positionAlign.html">/webvtt/api/VTTCue/positionAlign.html</a></summary>


|                  Subtest                   | Results |
|--------------------------------------------|---------|
|                                            | OK      |
| `VTTCue.positionAlign, script-created cue` | FAIL    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4921/webvtt/api/VTTCue/region.html">/webvtt/api/VTTCue/region.html</a></summary>


|               Subtest               | Results |
|-------------------------------------|---------|
|                                     | OK      |
| `VTTCue.region, script-created cue` | FAIL    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4921/webvtt/api/VTTCue/constructor.html">/webvtt/api/VTTCue/constructor.html</a></summary>


|           Subtest           | Results |
|-----------------------------|---------|
|                             | OK      |
| `VTTCue(), initial values`  | FAIL    |
| `VTTCue(), bad end time`    | PASS    |
| `VTTCue(), bad start time`  | PASS    |
| `VTTCue(), text formatting` | PASS    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4921/webvtt/api/VTTCue/line.html">/webvtt/api/VTTCue/line.html</a></summary>


|              Subtest              | Results |
|-----------------------------------|---------|
|                                   | OK      |
| `VTTCue.line, parsed cue`         | FAIL    |
| `VTTCue.line, script-created cue` | PASS    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4921/webvtt/api/VTTCue/position.html">/webvtt/api/VTTCue/position.html</a></summary>


|                Subtest                | Results |
|---------------------------------------|---------|
|                                       | OK      |
| `VTTCue.position, script-created cue` | FAIL    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4921/webvtt/api/VTTCue/snapToLines.html">/webvtt/api/VTTCue/snapToLines.html</a></summary>


|                 Subtest                  | Results |
|------------------------------------------|---------|
|                                          | OK      |
| `VTTCue.snapToLines, script-created cue` | PASS    |
| `VTTCue.snapToLines, parsed cue`         | FAIL    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4921/webvtt/api/VTTCue/size.html">/webvtt/api/VTTCue/size.html</a></summary>


|              Subtest              | Results |
|-----------------------------------|---------|
|                                   | OK      |
| `VTTCue.size, script-created cue` | FAIL    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4921/webvtt/api/VTTCue/align.html">/webvtt/api/VTTCue/align.html</a></summary>


|              Subtest               | Results |
|------------------------------------|---------|
|                                    | OK      |
| `VTTCue.align, parsed cue`         | FAIL    |
| `VTTCue.align, script-created cue` | FAIL    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4921/webvtt/api/VTTCue/lineAlign.html">/webvtt/api/VTTCue/lineAlign.html</a></summary>


|                Subtest                 | Results |
|----------------------------------------|---------|
|                                        | OK      |
| `VTTCue.lineAlign, script-created cue` | FAIL    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4921/webvtt/api/VTTCue/vertical.html">/webvtt/api/VTTCue/vertical.html</a></summary>


|                Subtest                | Results |
|---------------------------------------|---------|
|                                       | OK      |
| `VTTCue.vertical, parsed cue`         | FAIL    |
| `VTTCue.vertical, script-created cue` | PASS    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4921/webvtt/api/VTTCue/text.html">/webvtt/api/VTTCue/text.html</a></summary>


|              Subtest              | Results |
|-----------------------------------|---------|
|                                   | OK      |
| `VTTCue.text, script-created cue` | PASS    |
| `VTTCue.text, parsed cue`         | FAIL    |

</details>
</details>

**Comment by w3c-bots** [comment]
# Firefox (nightly channel) #
Testing web-platform-tests at revision fa67745f15d6223bf2b05fa0c40449196f724aef
Using browser at version BuildID 20170123125947; SourceStamp 36486fdc3813ef7943ae5b07b4128866d1938a6c
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>12 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/4921/webvtt/api/VTTCue/getCueAsHTML.html">/webvtt/api/VTTCue/getCueAsHTML.html</a></summary>


|                  Subtest                  | Results |
|-------------------------------------------|---------|
|                                           | OK      |
| `VTTCue.getCueAsHTML(), <c>`              | FAIL    |
| `VTTCue.getCueAsHTML(), creating the cue` | FAIL    |
| `VTTCue.getCueAsHTML(), <b>`              | FAIL    |
| `VTTCue.getCueAsHTML(), <u>`              | FAIL    |
| `VTTCue.getCueAsHTML(), <rt>`             | FAIL    |
| `VTTCue.getCueAsHTML(), <v>`              | FAIL    |
| `VTTCue.getCueAsHTML(), <i>`              | FAIL    |
| `VTTCue.getCueAsHTML(), x\0`              | FAIL    |
| `VTTCue.getCueAsHTML(), <ruby>`           | FAIL    |
| `VTTCue.getCueAsHTML(), <c.a.b>`          | FAIL    |
| `VTTCue.getCueAsHTML(), <v a b>`          | FAIL    |
| `VTTCue.getCueAsHTML(), <00:00:00.500>`   | FAIL    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4921/webvtt/api/VTTCue/positionAlign.html">/webvtt/api/VTTCue/positionAlign.html</a></summary>


|                  Subtest                   | Results |
|--------------------------------------------|---------|
|                                            | OK      |
| `VTTCue.positionAlign, script-created cue` | PASS    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4921/webvtt/api/VTTCue/region.html">/webvtt/api/VTTCue/region.html</a></summary>


|               Subtest               | Results |
|-------------------------------------|---------|
|                                     | OK      |
| `VTTCue.region, script-created cue` | FAIL    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4921/webvtt/api/VTTCue/constructor.html">/webvtt/api/VTTCue/constructor.html</a></summary>


|           Subtest           | Results |
|-----------------------------|---------|
|                             | OK      |
| `VTTCue(), initial values`  | FAIL    |
| `VTTCue(), bad end time`    | PASS    |
| `VTTCue(), bad start time`  | PASS    |
| `VTTCue(), text formatting` | FAIL    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4921/webvtt/api/VTTCue/line.html">/webvtt/api/VTTCue/line.html</a></summary>


|              Subtest              | Results |
|-----------------------------------|---------|
|                                   | OK      |
| `VTTCue.line, parsed cue`         | PASS    |
| `VTTCue.line, script-created cue` | PASS    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4921/webvtt/api/VTTCue/position.html">/webvtt/api/VTTCue/position.html</a></summary>


|                Subtest                | Results |
|---------------------------------------|---------|
|                                       | OK      |
| `VTTCue.position, script-created cue` | FAIL    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4921/webvtt/api/VTTCue/snapToLines.html">/webvtt/api/VTTCue/snapToLines.html</a></summary>


|                 Subtest                  | Results |
|------------------------------------------|---------|
|                                          | OK      |
| `VTTCue.snapToLines, script-created cue` | PASS    |
| `VTTCue.snapToLines, parsed cue`         | PASS    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4921/webvtt/api/VTTCue/size.html">/webvtt/api/VTTCue/size.html</a></summary>


|              Subtest              | Results |
|-----------------------------------|---------|
|                                   | OK      |
| `VTTCue.size, script-created cue` | FAIL    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4921/webvtt/api/VTTCue/align.html">/webvtt/api/VTTCue/align.html</a></summary>


|              Subtest               | Results |
|------------------------------------|---------|
|                                    | OK      |
| `VTTCue.align, parsed cue`         | PASS    |
| `VTTCue.align, script-created cue` | PASS    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4921/webvtt/api/VTTCue/lineAlign.html">/webvtt/api/VTTCue/lineAlign.html</a></summary>


|                Subtest                 | Results |
|----------------------------------------|---------|
|                                        | OK      |
| `VTTCue.lineAlign, script-created cue` | PASS    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4921/webvtt/api/VTTCue/vertical.html">/webvtt/api/VTTCue/vertical.html</a></summary>


|                Subtest                | Results |
|---------------------------------------|---------|
|                                       | OK      |
| `VTTCue.vertical, parsed cue`         | PASS    |
| `VTTCue.vertical, script-created cue` | PASS    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4921/webvtt/api/VTTCue/text.html">/webvtt/api/VTTCue/text.html</a></summary>


|              Subtest              | Results |
|-----------------------------------|---------|
|                                   | OK      |
| `VTTCue.text, script-created cue` | PASS    |
| `VTTCue.text, parsed cue`         | PASS    |

</details>
</details>

**Comment by BenjaminSchaaf** [comment]
I also have a full set of tests for the `VTTRegion` interface, but I'm still waiting on confirmation on some of the semantics before committing.

File: webvtt/api/VTTCue/positionAlign.html [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    var cue = new VTTCue(0, 1, 'text');
+
+    ['line-left', 'center', 'line-right', 'auto'].forEach(function(valid) {
+        cue.positionAlign = valid;
+        assert_equals(cue.positionAlign, valid);
+    });
+
+    cue.positionAlign = 'center';
+    ['auto\u0000', 'centre', 'middle'].forEach(function(invalid) {
+        cue.positionAlign = invalid
```
Comment (silviapfeiffer) [inline webvtt/api/VTTCue/positionAlign.html]: missing semicolon

**Review by silviapfeiffer — APPROVED** [review]
LGTM

**Comment by BenjaminSchaaf** [comment]
I've fixed up some of the `VTTCue` tests and committed the `VTTRegion` tests.

File: webvtt/api/VTTRegion/lines.html [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        region.lines = i;
+        assert_equals(region.lines, i);
+    }
+
+    [0, -1, 101, 4294967296, 18446744073709552000,
+     10000000000000000000000000000000000].forEach(function (valid) {
+        region.lines = valid;
+        assert_equals(region.lines, valid);
+    });
+}, document.title + ' initial values');
```
Comment (zcorpan) [inline webvtt/api/VTTRegion/lines.html]: Should say "script-created region" or so? Or could remove this argument, it defaults to `document.title` if not given.

Same thing in subsequent tests below.

**Review by zcorpan — APPROVED** [review]
LGTM % nit about test names

File: webvtt/api/VTTRegion/lines.html (hunk 2 above)
Comment (BenjaminSchaaf) [inline webvtt/api/VTTRegion/lines.html]: Yes it should :)

**Comment by zcorpan** [comment]
/webvtt/api/VTTCue/align.html "middle" -> "center" was fixed just recently in
https://bugs.chromium.org/p/chromium/issues/detail?id=663797 (test passes in 58.0.3020.0 canary)

**Comment by zcorpan** [comment]
/webvtt/api/VTTCue/constructor.html `VTTCue.region` was implemented in
https://bugs.chromium.org/p/chromium/issues/detail?id=690014

`lineAlign` and `positionAlign`: https://bugs.chromium.org/p/chromium/issues/detail?id=633690

**Comment by zcorpan** [comment]
Bugs to notify that these tests moved:

https://bugzilla.mozilla.org/show_bug.cgi?id=1341568
https://bugs.chromium.org/p/chromium/issues/detail?id=694960

**Pull request merged**